### PR TITLE
Index creation support specify partition

### PR DESCRIPTION
**Pull request by xuanyuanking**
## What changes were proposed in this pull request?

#232 Add support in ASTBuilder and get partition by partitionSpec in SpinachUtils, re-implement for OAP

## How was this patch tested?

Add new test case in SpinachDDLSuite, check the index file exists or not

### CODE REVIEW

**Comment by xuanyuanking** [comment]
Fix the bug in #233 and add the index file check in UT.
@adrian-wang Please review this, thanks.

File: src/main/scala/org/apache/spark/sql/execution/datasources/spinach/index/indexPlans.scala [hunk 1]
```diff
@@ -121,8 +123,11 @@ case class CreateIndex(
     val ids =
       indexColumns.map(c => s.map(_.name).toIndexedSeq.indexOf(c.columnName))
     val keySchema = StructType(ids.map(s.toIndexedSeq(_)))
-    val queryExecution =
-      Dataset.ofRows(sparkSession, Project(ids.map(relation.output), relation)).queryExecution
+    var ds = Dataset.ofRows(sparkSession, Project(ids.map(relation.output), relation))
+    partitionSpec.get.foreach { case (k, v) =>
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/spinach/index/indexPlans.scala]: Can you check if `partitionSpec` is null?

Comment (xuanyuanking) [inline src/main/scala/org/apache/spark/sql/execution/datasources/spinach/index/indexPlans.scala]: Fixed done, sorry for ignoring the ci check.

**Comment by adrian-wang** [comment]
LGTM

**Pull request merged**